### PR TITLE
XWIKI-15956: Suggest Fails To Display Current Value Correctly On Edit

### PR DESCRIPTION
**Pull request by atallahade**
### Issue

https://jira.xwiki.org/browse/XWIKI-15956

### Changes

* Implement the getValue method for DBList classes.

### CODE REVIEW

**Comment by vmassol** [comment]
Thanks @atallahade  that was fast :) Is there some "easy" way to have some tests for this (be them unit or functional)?

**Comment by atallahade** [comment]
@vmassol I'll add some unit tests.

File: xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/DBListClassPropertyValuesProvider.java [hunk 1]
```diff
@@ -61,4 +64,26 @@ protected PropertyValues getAllowedValues(DBListClass dbListClass, int limit, St
...
+                                dbListClass.getDocumentReference());
+            } catch (Exception e) {
+                throw new XWikiRestException(e);
+            }
+        }
+        if (propertyValue == null) {
+            propertyValue = super.getValue(propertyReference, rawValue);
+        }
+
+        return propertyValue;
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/DBListClassPropertyValuesProvider.java]: I find this overly complicated. Why not simply call

    getValues(propertyReference, 1, rawValue.toString())

and return the first value from the returned ``PropertyValues``?

Comment (atallahade) [reply xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/DBListClassPropertyValuesProvider.java]: @mflorea The query in `#getValues` uses this text filter: `text like '%value%'`, hence the first value returned by `#getValues` might not be the one that we are looking for. What we want is `text = 'value'`. (or `text like 'value'`)

Comment (mflorea) [reply xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/DBListClassPropertyValuesProvider.java]: @atallahade right, it looks good then.

**Pull request merged**